src/main/java/pl/stormit/tinyurl/service/IpLocalizationService.java

```diff
@@ -0,0 +1,42 @@
+package pl.stormit.tinyurl.service;
+
+import com.maxmind.geoip2.DatabaseReader;
+import com.maxmind.geoip2.exception.AddressNotFoundException;
+import com.maxmind.geoip2.exception.GeoIp2Exception;
+import com.maxmind.geoip2.model.CityResponse;
+import org.springframework.stereotype.Service;
+import pl.stormit.tinyurl.dto.UrlAnalyticsLocalizationDto;
+
+import java.io.File;
+import java.io.IOException;
+import java.net.InetAddress;
+
+@Service
+public class IpLocalizationService {
+
+    public UrlAnalyticsLocalizationDto getIpLocalization(String addressIp) {
+
+        InetAddress ipAddress;
+        UrlAnalyticsLocalizationDto setLocalization = new UrlAnalyticsLocalizationDto(null, null, null);
```

Comment:
3 x null, looks really strange.
Maybe it will be better to use some default values, for example in one of the constructors.

Reply:
At first, I initialized fields country, isoCode, city with empty strings.
Later we will may be change it according to the requirements of your program.